src/react/components/common/assetPreview/tfrecordAsset.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import { IAssetProps } from "./assetPreview";
+import { IAsset, AssetType } from "../../../../models/applicationState";
+import HtmlFileReader from "../../../../common/htmlFileReader";
+import { TFRecordsReader } from "../../../../providers/export/tensorFlowRecords/tensorFlowReader";
+import { FeatureType } from "../../../../providers/export/tensorFlowRecords/tensorFlowBuilder";
+
+/**
+ * Properties for TFRecord Asset Image component
+ */
+export interface ITFRecordProps extends IAssetProps, React.ClassAttributes<TFRecordAsset> {
+}
+
+/**
+ * State for TFRecord Asset Image component
+ * @member tfRecordImage64 - base64 representation of the image data
+ */
+export interface ITFRecordState {
+    tfRecordImage64: string;
+}
+
+/**
+ * React component that displays an image from a TFRecord asset file
+ */
+export class TFRecordAsset extends React.Component<ITFRecordProps, ITFRecordState> {
+    public state: ITFRecordState = {
+        tfRecordImage64: "",
+    };
+
+    private image: React.RefObject<HTMLImageElement> = React.createRef();
+
+    public render() {
```

Comment:
Instead of re-implementing the logic from the `ImageAsset` you should just reuse the image asset and bind a newly constructed asset object.  You can set your base64 data uri as the path in the generated asset.